### PR TITLE
configuration_manual/nfs: Rewrite the page

### PR DESCRIPTION
**Pull request by sirainen**
Remove things that are obsolete or really not recommended at all.
Also remove suggestions that aren't clearly explained.

### CODE REVIEW

File: source/configuration_manual/nfs.rst [hunk 1]
```diff
@@ -4,178 +4,109 @@
...
-NFS is commonly used in one of these ways:
-
-1. Dovecot is run in a single computer.
-
-2. Dovecot is run in multiple computers, users are redirected more or
-   less randomly to different computers.
+Dovecot is commonly used with NFS. However, Dovecot does **not** support
+accessing the same user simultaneously by different servers. That will
+result in more or less severe mailbox corruption. Note that this applies
+to all kinds of mailbox access, including mail deliveries.
```
Comment (thuovila) [inline source/configuration_manual/nfs.rst:10]: I would remove "kinds of" and just say "applies to all mailbox access, including mail delivery."

File: source/configuration_manual/nfs.rst [hunk 2]
```diff
@@ -4,178 +4,109 @@
...
+=================
+
+* ``nordirplus``: Disable readdirplus operations, which aren't needed by
+  Dovecot. They can also slow down some NFS servers.
+
+* ``root_squash``: Dovecot doesn't care about this. Typically Dovecot doesn't
+  store any root-owned files in NFS.
+
+* ``nolock``: This is possible to use either if ``lock_method=dotlock`` is
+  used, 2or as a slightly unsafe optimization (see below).
```
Comment (thuovila) [inline source/configuration_manual/nfs.rst:34]: s/2or/or ?

File: source/configuration_manual/nfs.rst [hunk 3]
```diff
@@ -4,178 +4,109 @@
...
+  accessed by multiple servers simultaneously. In those situations the mails
+  are more likely to become corrupted if ``nolock`` is used.
 
+Clock synchronization
+=====================
 
-.. note::
+Run ntpd in the NFS server and all the NFS clients to make sure their
+clocks are synchronized. They should always be less than 1 second apart
+from each others.
```
Comment (thuovila) [inline source/configuration_manual/nfs.rst:74]: from each other

File: source/configuration_manual/nfs.rst [hunk 4]
```diff
@@ -4,178 +4,109 @@
...
-   mail_nfs_storage = no
-   mail_nfs_index = no
+* Set :ref:`setting-mail_nfs_index` = ``yes`` and
+  :ref:`setting-mail_nfs_storage` = ``yes``. These will attempt to flush the NFS
+  caches at appropriate times. However, it doesn't work perfectly.
 
-Multi-server setup that **tries** to flush NFS caches (increases NFS
-operations, and **isn't fully reliable**), try not to use this:
+    * Disabling NFS attribute cache helps a lot in getting rid of caching
+      related errors, but this makes the performance MUCH worse and increases
```
Comment (thuovila) [inline source/configuration_manual/nfs.rst:95]: Not sure, but I think it is just "performance" without "the". E.g.in the next paragraph it is used that way and sounds nicer.

**Review by thuovila — APPROVED** [review]

**Review by cmouse — APPROVED** [review]

**Pull request merged**